arch/arm/core/aarch32/swap_helper.S

```diff
@@ -498,8 +498,29 @@ _stack_frame_endif:
 
 _oops:
     push {r0, lr}
+#if defined(CONFIG_EXTRA_EXCEPTION_INFO)
+#if defined(CONFIG_ARMV7_M_ARMV8_M_MAINLINE)
+    /* Build _callee_saved_t. To match the struct
+     * definition we push the psp & then r11-r4
+     */
+    mrs r1, PSP
+    push {r1}
```

Comment:
You aren't restoring r1 after the call to z_do_kernel_oops.  Is there a reason to stack this at all?

Reply:
PSP is the last member of _callee_saved_t, I'm pushing it on the stack so that the struct passed to z_do_kernel_oops contains it as expected